make_a_plea/management/commands/zendesk.py

```diff
@@ -0,0 +1,212 @@
+        self.zendesk = Zendesk(
+            self.settings["url"],
+            self.settings["email"],
+            self.settings["password"])
+
+    def get_all_fieldnames(self):
+        return \
+            SIMPLE_FIELDS + \
+            NEWLINE_FIELDS + \
+            STRUCTURED_FIELDS
+
+    def to_csv(self):
+        self.outfile = self.settings.get('outfile', 'zendump.csv')
+        self.batch_size = 100
+        with open(self.outfile, "wb") as f:
+
+            self.init_csv(f)
+            self.writerow(self.get_all_fieldnames())  # CSV Header
+
+            for batch in self.yield_ticket_batches():
+                for ticket in batch:
+                    self.writerow(self.ticket_to_row(ticket))
+
+    def writerow(self, row):
+        self.writer.writerow([
+            cell.encode("utf-8")
+            for cell in row])
+
+    def ticket_to_row(self, ticket):
+        return \
+            self.get_simple_fields(ticket) + \
+            self.get_newline_fields(ticket) + \
+            self.get_structured_fields(ticket)
+
+    def parse_simple_field(self, field):
+        try:
+            return field.encode("utf-8")
+        except AttributeError:  # ints
+            return str(field).encode("utf-8")
+
+    def get_simple_fields(self, ticket):
+        fields = []
+        for field in SIMPLE_FIELDS:
+            if field in ticket:
+                fields.append(self.parse_simple_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
+
+    def parse_newline_field(self, field):
+        return field or ""
+
+    def get_newline_fields(self, ticket):
+        fields = []
+        for field in NEWLINE_FIELDS:
+            if field in ticket and field is not None:
+                fields.append(self.parse_newline_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
+
+    def parse_structured_field(self, field):
+        return str(field)
+
+    def get_structured_fields(self, ticket):
+        fields = []
+        for field in STRUCTURED_FIELDS:
+            if field in ticket:
+                fields.append(self.parse_structured_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
+
+    def yield_ticket_batches(self):
+        """Generates batches of tickets"""
+
+        for raw_query in self.settings["raw_queries"]:
+
+            first_page = self.zendesk.search(raw_query=raw_query)
+            yield first_page["results"]
+
+            count = first_page["count"]
+            print "{} tickets from ZenDesk match filter.".format(count)
+            page_count, remainder = divmod(count, self.batch_size)
+            page_count = page_count + 1 if remainder else page_count
+
+            for page_id in range(1, page_count):
+                qpage = " page:{}".format(page_id)
+                page = self.zendesk.search(raw_query=raw_query + qpage)
+                yield page["results"]
```

Comment:
Not a biggie but, if you loop over the results in here and yield each one you could hide the fact that this it chunked from the caller.